tests/rkt_image_dependencies_test.go

```diff
@@ -0,0 +1,220 @@
+// Copyright 2015 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+
+	taas "github.com/coreos/rkt/tests/test-auth-server/aci"
+	"github.com/coreos/rkt/tests/testutils"
+)
+
+const (
+	manifestDepsTemplate = `
+{
+   "acKind" : "ImageManifest",
+   "acVersion" : "0.7.3",
+   "dependencies" : [
+      DEPENDENCIES
+   ],
+   "labels" : [
+      {
+         "name" : "version",
+         "value" : "VERSION"
+      },
+      {
+         "name" : "arch",
+         "value" : "amd64"
+      },
+      {
+         "value" : "linux",
+         "name" : "os"
+      }
+   ],
+   "app" : {
+      "user" : "0",
+      "exec" : [
+         "/inspect", "--print-msg=HelloDependencies"
+      ],
+      "workingDirectory" : "/",
+      "group" : "0",
+      "environment" : [
+      ]
+   },
+   "name" : "IMG_NAME"
+}
+`
+)
+
+// TestImageDependencies generates ACIs with a complex dependency tree and
+// fetches them via the discovery mechanism. Some dependencies are already
+// cached in the CAS, and some dependencies are fetched via the discovery
+// mechanism. This is to reproduce the scenario in explained in:
+// https://github.com/coreos/rkt/issues/1752#issue-117121841
+func TestImageDependencies(t *testing.T) {
+	tmpDir := createTempDirOrPanic("rkt-TestImageDeps-")
+	defer os.RemoveAll(tmpDir)
+
+	ctx := testutils.NewRktRunCtx()
+	defer ctx.Cleanup()
+
+	// TODO: we can avoid setting the port manually when appc/spec gains
+	// the ability to specify ports for discovery.
+	// See https://github.com/appc/spec/pull/110
+	//
+	// httptest by default uses random ports. We override this via the
+	// "httptest.serve" flag.
+	//
+	// As long as we set the port via the "httptest.serve" flag, we have
+	// to use https rather than http because httptest.Start() would wait
+	// forever in "select {}", see
+	// https://golang.org/src/net/http/httptest/server.go?s=2768:2792#L92
+	//
+	// This means this test must:
+	// - use https only
+	// - ignore tls errors with --insecure-options=tls
+	serverURL := flag.Lookup("httptest.serve")
```

Comment:
Maybe handle the case, when serverURL is `nil`. Like panic when it is nil, with some useful message instead of getting a nil dereference error.